x/cronos/spec/04_messages.md

```diff
@@ -0,0 +1,36 @@
+<!-- order: 4 -->
+
+# Messages
+
+## MsgConvertVouchers
+
+Convert native tokens to the mapped CRC20 tokens, if the mapping does not exist and auto-deployment is enabled, an embed CRC20 contract is deployed for it automatically, otherwise, the message fails.
+
++++ https://github.com/crypto-org-chain/cronos/blob/v0.6.0-testnet/proto/cronos/tx.proto#L26-L30
+
+This message is expected to fail if:
+
+- The coin denom is neither IBC nor gravity tokens.
+- The mapping does not exist and auto-deployment is not enabled.
+
+## MsgTransferTokens
+
+Transfer IBC tokens away from Cronos chain, will do decimals conversion automatically for CRO.
+
++++ https://github.com/crypto-org-chain/cronos/blob/v0.6.0-testnet/proto/cronos/tx.proto#L33-L38
+
+This message is expected to fail if:
+
+- The sender doesn't have enough balance.
+- The IBC transfer message fails.
+
+## MsgUpdateTokenMapping
+
+Update external token mapping, insert if not exists, can only be called by Cronos admin account, which is configured in module parameters.
```

Comment:
how is the supply managed if there was an existing token mapping, and it was updated to a different one?